### PR TITLE
Add configuration for fields which need to be highlighted in the image

### PR DESCRIPTION
**Pull request by beatrycze-volk**
This PR allows to use more than one field for highlighting words in the image. It gives also more flexibility in the name of parameters. 
Example: use 'query' param to highlight word after clicking the found result in the result list, right after chosen document is loaded it highlights word for which user was searching and then you have param 'highlight_word' which highlights word searched already inside the document itself. The first defined param on the configuration has highest priority, if not found it checks next ones. Parameters are separated by commas.

- add TypoScript configuration
- add documentation for configurable field 'searchHlElements'
- pass field value to full text control by dictionary
- use it in page view

### CODE REVIEW

File: Configuration/TypoScript/Toolbox/setup.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 plugin.tx_dlf_annotationtool.templateFile = EXT:dlf/Resources/Private/Templates/AnnotationTool.tmpl
 plugin.tx_dlf_fulltexttool.activateFullTextInitially = 0
 plugin.tx_dlf_fulltexttool.fullTextScrollElement = html, body
+plugin.tx_dlf_fulltexttool.searchHlElements = tx_dlf[highlight_word]
```
Comment (sebastian-meyer) [inline Configuration/TypoScript/Toolbox/setup.txt]:
```suggestion
plugin.tx_dlf_fulltexttool.searchHlParameters = tx_dlf[highlight_word]
```

File: Documentation/Plugins/Index.rst [hunk 2]
```diff
@@ -1037,7 +1037,9 @@ Fulltext Tool
 ^^^^^^^^^^^^^
 This plugin adds an activation link for fulltext to the toolbox. If no fulltext is available for the current page, a span-tag is rendered instead.
 
-The default behaviour is to show the fulltext after click on the toggle link. There is a TypoScript configuration to show the fulltext initially.
+The default behavior is to show the fulltext after click on the toggle link. There is a TypoScript configuration to show the fulltext initially.
+
+Plugin allows also to configure (searchHlElements) by which URL parameters words will be highlighted in the image. The first defined parameter on the configuration has highest priority, if not found it checks the next ones.
```
Comment (sebastian-meyer) [inline Documentation/Plugins/Index.rst]:
```suggestion
Plugin allows also to configure (searchHlParameters) by which URL parameters words will be highlighted in the image. The first defined parameter on the configuration has highest priority, if not found it checks the next ones.
```

File: Documentation/Plugins/Index.rst [hunk 3]
```diff
@@ -1069,9 +1071,16 @@ The default behaviour is to show the fulltext after click on the toggle link. Th
    :Data Type:
        :ref:`t3tsref:data-type-string`
    :Default:
-        html, body       
+        html, body
+
+ - :Property:
+       searchHlElements
```
Comment (sebastian-meyer) [inline Documentation/Plugins/Index.rst]:
```suggestion
       searchHlParameters
```

File: Classes/Plugin/Tools/FulltextTool.php [hunk 4]
```diff
@@ -77,7 +77,13 @@ public function main($content, $conf)
...
+            $markerArray['###FULLTEXT_SELECT###'] = '<a class="select switchoff" id="tx-dlf-tools-fulltext" title="" data-dic="'
+            . 'fulltext:' . htmlspecialchars($this->pi_getLL('fulltext', ''))
+            . ';fulltext-on:' . htmlspecialchars($this->pi_getLL('fulltext-on', ''))
+            . ';fulltext-off:' . htmlspecialchars($this->pi_getLL('fulltext-off', ''))
+            . ';activate-full-text-initially:' . MathUtility::forceIntegerInRange($this->conf['activateFullTextInitially'], 0, 1, 0)
+            . ';full-text-scroll-element:' . $this->conf['fullTextScrollElement']
+            . ';search-hl-elements:' . $this->conf['searchHlElements'] . '">&nbsp;</a>';
```
Comment (sebastian-meyer) [inline Classes/Plugin/Tools/FulltextTool.php]:
```suggestion
            . ';search-hl-parameters:' . $this->conf['searchHlParameters'] . '">&nbsp;</a>';
```

File: Resources/Public/Javascript/PageView/FulltextControl.js [hunk 5]
```diff
@@ -57,7 +57,13 @@ var dlfViewerFullTextControl = function(map, image, fulltextUrl) {
...
     this.dic = $('#tx-dlf-tools-fulltext').length > 0 && $('#tx-dlf-tools-fulltext').attr('data-dic') ?
         dlfUtils.parseDataDic($('#tx-dlf-tools-fulltext')) :
-        {'fulltext':'Fulltext', 'fulltext-on':'Activate Fulltext','fulltext-off':'Deactivate Fulltext', 'activate-full-text-initially':'0', 'full-text-scroll-element':'html, body'};
+        {
+            'fulltext':'Fulltext',
+            'fulltext-on':'Activate Fulltext',
+            'fulltext-off':'Deactivate Fulltext',
+            'activate-full-text-initially':'0',
+            'full-text-scroll-element':'html, body',
+            'search-hl-elements':'tx_dlf[highlight_word]'};
```
Comment (sebastian-meyer) [inline Resources/Public/Javascript/PageView/FulltextControl.js]:
```suggestion
            'search-hl-parameters':'tx_dlf[highlight_word]'};
```

File: Resources/Public/Javascript/PageView/FulltextControl.js [hunk 6]
```diff
@@ -70,6 +76,12 @@ var dlfViewerFullTextControl = function(map, image, fulltextUrl) {
      * @private
      */
     this.fullTextScrollElement = this.dic['full-text-scroll-element'];
+
+    /**
+     * @type {string}
+     * @private
+     */
+    this.searchHlElements = this.dic['search-hl-elements'];
```
Comment (sebastian-meyer) [inline Resources/Public/Javascript/PageView/FulltextControl.js]:
```suggestion
    this.searchHlParameters = this.dic['search-hl-parameters'];
```

File: Resources/Public/Javascript/PageView/PageView.js [hunk 7]
```diff
@@ -161,6 +167,7 @@ dlfViewer.prototype.addCustomControls = function(controlNames) {
     if (this.fulltexts[0] !== undefined && this.fulltexts[0].length !== 0 && this.fulltexts[0].url !== '' && this.images.length === 1) {
         fulltextControl = new dlfViewerFullTextControl(this.map, this.images[0], this.fulltexts[0].url);
         fulltextDownloadControl = new dlfViewerFullTextDownloadControl(this.map, this.images[0], this.fulltexts[0].url);
+        this.highlightKeys = fulltextControl.searchHlElements;
```
Comment (sebastian-meyer) [inline Resources/Public/Javascript/PageView/PageView.js]:
```suggestion
        this.highlightKeys = fulltextControl.searchHlParameters;
```

**Review by sebastian-meyer — CHANGES_REQUESTED** [review]
I suggest calling the option `searchHlParameters` instead since we are dealing with URL parameters here. `elements` usually refers to HTML tags.

**Review by albig — APPROVED** [review]
I checked again the functionality and it works with the FulltextTool as described.

A little confusing was in my setup that the word highlighting is working without the FulltextTool, too. In this case the setting of `search-hl-parameters` is fixed to `tx_dlf[highlight_word]`.

I would merge this tomorrow.

**Review by sebastian-meyer — APPROVED** [review]

**Pull request merged**